### ISSUE TITLE
[Bug]: open metamask with deeplink in andoird chrome, mm browser will open two different website 

### ISSUE THREAD

**Issue by hellofriday** [MetaMask/metamask-mobile#7724]
### Describe the bug

Android device
mm version is v7.9.1

In the Chrome browser, 
I'm using a deep link to open a website with MetaMask. 
However, MetaMask's browser is opening two different websites simultaneously. 
The one on top is a website I previously closed, and the one below is the website I intend to open this time.

Only after killing the MetaMask android app process, the above operation performs normally in MetaMask.

In IOS every thing is fine。

### Expected behavior

Only open one website in mm browser。

### Screenshots/Recordings

https://github.com/MetaMask/metamask-mobile/assets/7859030/38dca3e4-ad8b-47f4-a5c6-2cc98da7f6cf

### Steps to reproduce

1. open a deeplink to sign message
2. close all tab in mm browser
3. open a another deeplink to sign transaction

### Version

7.9.1

### Build type

None

### Device

xiaomi/samsung

### Operating system

Android

**Comment by omridan159** [comment]
Hey @hellofriday 
Could you pls provide us with the exact deeplink that you call in your app so we can try and reproduce it?

Thanks!

**Comment by ada20221024** [comment]
@omridan159 
 sign message deeplink :
dapp://app.crossspace.io/async/async-sign?sid=52390eb9-858b-44c0-b199-f9e0e92268db

trade deeplink:
dapp://app.crossspace.io/async/async-trade-keys?address=0x655008151f2dbeb52029Cf051bfaFa0ca7f479c4&abi=tradeAbi&function_name=sellShares&args=%5B%220xA30Eb0e0e2B632Df08a4328604be2fBbFe54B35d%22%2C%226551d203264ad02968234b30%22%2C1%5D&value=0&contractNetwork=80001&feedID=6551d203264ad02968234b30&orderInfo=%7B%22ticketPrice%22%3A%2218000000000000000%22%2C%22creatorKeyQuantity%22%3A%22382556488862933290%22%2C%22keyPrice%22%3A%228050466206443634%22%2C%22fee%22%3A%222605046620644362%22%2C%22totalPrice%22%3A%2223445419585799272%22%2C%22tradeType%22%3A%22sell%22%2C%22amount%22%3A1%7D&sid=6538cd4e-cf33-4602-9e04-1d9f60f3910e

**Comment by hellofriday** [comment]
@omridan159 

sign message deeplink :
dapp://app.crossspace.io/async/async-sign?sid=52390eb9-858b-44c0-b199-f9e0e92268db

trade deeplink:
dapp://app.crossspace.io/async/async-trade-keys?address=0x655008151f2dbeb52029Cf051bfaFa0ca7f479c4&abi=tradeAbi&function_name=sellShares&args=%5B%220xA30Eb0e0e2B632Df08a4328604be2fBbFe54B35d%22%2C%226551d203264ad02968234b30%22%2C1%5D&value=0&contractNetwork=80001&feedID=6551d203264ad02968234b30&orderInfo=%7B%22ticketPrice%22%3A%2218000000000000000%22%2C%22creatorKeyQuantity%22%3A%22382556488862933290%22%2C%22keyPrice%22%3A%228050466206443634%22%2C%22fee%22%3A%222605046620644362%22%2C%22totalPrice%22%3A%2223445419585799272%22%2C%22tradeType%22%3A%22sell%22%2C%22amount%22%3A1%7D&sid=6538cd4e-cf33-4602-9e04-1d9f60f3910e

thx!

**Comment by christopherferreira9** [comment]
HI @hellofriday 
Are you trying to open both links with one tap? Every `dapp://...` will open a new mm browser tab. 
We suggest using the MetaMask SDK as it is the best solution to handle connections between dapps and the MetaMask Mobile wallet. You can find the docs [here](https://docs.metamask.io/sdk) and the github repository [here](https://github.com/MetaMask/metamask-sdk).

**Comment by hellofriday** [comment]
@christopherferreira9 
No, what I mean is to only open one link at a time. But when I open a second link after a while, I find that MM Tab mysteriously opens two tabs at the same time.

**Comment by hellofriday** [comment]
Also, I'm not sure if you have done any testing on Android. We initially had issues when integrating with the SDK, which is another problem. Eventually, we chose to integrate using deep links.

**Comment by tommasini** [comment]
Hey @hellofriday, thanks for sharing that!

Could you create an issue with the issues you had integrating with the SDK? We would love to have your feedback! Thanks!

**Comment by hellofriday** [comment]
@tommasini 
Ok
but, can you fix this deeplink issue first, it block our users，thanks.

**Comment by davidliudev** [comment]
hihi @tommasini @christopherferreira9 thanks for your attention on this. we will raise another issue soon regarding the SDK.
our team will try to assist on the replication of this issues.

meanwhile regarding this issue, @hellofriday please provide as much information as helpful so the metamask team could help investigate.

**Comment by hellofriday** [comment]
@tommasini 
create an issue with the issues you had integrating with the SDK 
 https://github.com/MetaMask/metamask-mobile/issues/7894
 
 @davidliudev  cc

**Comment by christopherferreira9** [comment]
Hi @hellofriday ! This seems to be fixed in 7.11. Can you please check the steps in the next video to see if I missed something?

https://github.com/MetaMask/metamask-mobile/assets/104831203/3718b57c-c2d3-4865-bcfa-9a04208e7c6f

**Comment by hellofriday** [comment]
Thx, the latest version on Play Store is still 7.10.0

**Comment by hellofriday** [comment]
![image](https://github.com/MetaMask/metamask-mobile/assets/7859030/012f6ada-35cb-4de9-81b5-73ca495d1a2d)

**Comment by A-Feder** [comment]
Hey @hellofriday, can you let me know if 7.11 is now available to you?

**Comment by hellofriday** [comment]
@A-Feder no， It still doesn't work on my Xiaomi 11 real device.
You are using an emulator for demonstration instead of a real device. I have developed Android apps for 3 years before, and I suggest testing such functional issues on a real device.
I have wirte a demo  and deploy in https://demo-seven-roan.vercel.app/demo，you can view the source code in https://github.com/hellofriday/demo/blob/main/src/Demo.tsx.

below is the video record to show this bug in 7.11.0 version.
Step:
1. click sign message button
2. click pay button

https://github.com/MetaMask/metamask-mobile/assets/7859030/37ed5aeb-7e95-4c0f-afa9-1a910838666b

**Comment by christopherferreira9** [comment]
Hi @hellofriday ! I will give that a try as soon as possible. 
The video I shared is from a real device, just using a screen capture app instead of recording the screen itself.

**Comment by christopherferreira9** [comment]
Hi @hellofriday . I gave it a try with the same dapp setup using the latest version of the MetaMask wallet `7.14.0` and my Pixel 7a (real, physical device) and wasn't able to reproduce it. Will check with the team and their own devices to see if we manage to reproduce it.

https://github.com/MetaMask/metamask-mobile/assets/104831203/18624086-1ad2-4111-815d-00d0ad3c3da5
